sql/core/src/main/scala/org/apache/spark/sql/execution/debug/package.scala

```diff
@@ -107,12 +107,19 @@ package object debug {
    */
   def codegenStringSeq(plan: SparkPlan): Seq[(String, String, ByteCodeStats)] = {
     val codegenSubtrees = new collection.mutable.HashSet[WholeStageCodegenExec]()
-    plan transform {
-      case s: WholeStageCodegenExec =>
-        codegenSubtrees += s
-        s
-      case s => s
+
+    def findSubtrees(plan: SparkPlan): Unit = {
+      plan transform {
+        case s: WholeStageCodegenExec =>
+          codegenSubtrees += s
```

Comment:
The code below has the same issue?
https://github.com/apache/spark/blob/df2314b63aaf4992ac86ea0b68dae8554b066828/sql/core/src/test/scala/org/apache/spark/sql/BenchmarkQueryTest.scala#L65-L70